### DISCUSSION TITLE
Ext: Chat Filter; If amount of filtered chat messages is >x, mute that player, might be useful if word list isn't complete, and you still see some of their messages

### DISCUSSION THREAD

**Discussion by Sponsorn** [runelite/runelite#17021] [Ideas]
I have a chat filter set up, but I can't filter every word or combo, so would be nice to be able to set up if a player hits a certain amount of filtered messages from my chat filter, the player would be muted

Example I have regex set up for \d.*% and buy messages in the beginning, but some still get through

Maybe add a time limit, so if no new blocked messages are sent within a x time period, the "yellow cards" would disappear/reset, which might decrease the size of cache? I don't know how those things work

**Comment by IanBurgan** [comment]
I like this idea and wanted something similar. I could probably implement it and put up a PR. Looking at the code, there doesn't appear to be a way to add them to the actual ignore list. The implementation that makes sense to me would be to add the player to the list of filtered player names in the plugin's configuration after enough violations.

If there is interest and a chance that the PR can get merged, I can implement this. It looks like a [related PR](https://github.com/runelite/runelite/pull/16836) was closed in the past

**Reply by IanBurgan** [reply]
Added a PR here: https://github.com/runelite/runelite/pull/17063. It differs slightly in that the config option is that the player is "muted" after any of their messages are filtered out rather than having a count.
